src/main/java/org/apache/mesos/hdfs/config/SchedulerConf.java

```diff
@@ -1,10 +1,28 @@
 package org.apache.mesos.hdfs.config;
 
+import com.google.inject.Singleton;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
 
-public class SchedulerConf extends MainConf {
-  public SchedulerConf(Configuration conf, int configServerPort) {
-    super(conf, configServerPort);
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Properties;
+
+@Singleton
+public class SchedulerConf extends Configured {
+
+  public SchedulerConf(Configuration conf) {
+    setConf(conf);
+  }
+
+  public SchedulerConf() {
+    Properties props = System.getProperties();
```

Comment:
We use a tool for process supervisor that specifies a config file location. The name of the config file is automatically generated. Also that config file is put in place on build. So you would need to change the config, build the tar.gz, then deploy your tar.gz. If you could change the config file you can just change the bash script call to include a file from a known location.

I don't think it's a huge deal either way honestly. I wish there was a nicer way to do this like there is in typesafe config.